kiwi/storage/disk.py

```diff
@@ -502,9 +514,16 @@ def __del__(self):
         if self.storage_provider.is_loop() and self.is_mapped:
             log.info('Cleaning up %s instance', type(self).__name__)
             try:
-                Command.run(
-                    ['partx', '--delete', self.storage_provider.get_device()]
-                )
+                if self.partition_mapper == 'partx':
+                    Command.run(
+                        ['partx', '--delete', self.storage_provider.get_device()]
+                    )
+                else:
+                    for device_node in self.partition_map.values():
+                        Command.run(['dmsetup', 'remove', device_node])
+                    Command.run(
+                        ['kpartx', '-d', self.storage_provider.get_device()]
+                    )
```

Comment:
This condition should be flipped. Only when `kpartx` is used should we use `kpartx`/`dmsetup`. That makes it obvious that `kpartx` is the special case.

Reply:
ok good point

Reply:
done